plugin/src/software/aws/toolkits/eclipse/amazonq/lsp/editor/ActiveEditorChangeListener.java

```diff
@@ -95,26 +113,39 @@ private void handleEditorChange(final ITextEditor editor) {
         }, DEBOUNCE_DELAY_MS);
     }
 
-    private Map<String, Object> createActiveEditorParams(final ITextEditor editor) {
+    private Map<String, Object> createActiveEditorParams(final Object editor) {
         Map<String, Object> params = new HashMap<>();
         if (editor != null) {
-            Optional<String> fileUri = QEclipseEditorUtils.getOpenFileUri(editor.getEditorInput());
-            if (fileUri.isPresent()) {
-                Map<String, String> textDocument = new HashMap<>();
-                textDocument.put("uri", fileUri.get());
-                params.put("textDocument", textDocument);
-                QEclipseEditorUtils.getSelectionRange(editor).ifPresent(range -> {
-                    Map<String, Object> cursorState = new HashMap<>();
-                    cursorState.put("range", range);
-                    params.put("cursorState", cursorState);
-                });
+            if (editor instanceof ITextEditor textEditor) {
+                Optional<String> fileUri = QEclipseEditorUtils.getOpenFileUri(textEditor.getEditorInput());
+
+                if (fileUri.isPresent()) {
+                    Map<String, String> textDocument = new HashMap<>();
+                    textDocument.put("uri", fileUri.get());
+                    params.put("textDocument", textDocument);
+                    QEclipseEditorUtils.getSelectionRange(textEditor).ifPresent(range -> {
+                        Map<String, Object> cursorState = new HashMap<>();
+                        cursorState.put("range", range);
+                        params.put("cursorState", cursorState);
+                    });
+                }
+            } else if (editor instanceof IEditorPart editorPart) {
```

Comment:
this will break and is too wide of a check. We only want to do semantic handling when editor is AdtEditor. 
```
else if (sAdtEditor(part))
```